src/workerd/api/sockets.c++

```diff
@@ -66,15 +20,22 @@ jsg::Ref<Socket> connectImplNoOutputLock(
   // TODO: Validate `address` is well formed. Do I need to use `parseAddress` here or is there
   // a better way?
 
-  // Set up the connection. This is similar to HttpClientAdapter::connect.
+  // Set up the connection.
   auto headers = kj::heap<kj::HttpHeaders>(ioContext.getHeaderTable());
-  auto paf = kj::newPromiseAndFulfiller<kj::HttpClient::ConnectResponse>();
-  auto tunnel = kj::refcounted<TCPConnectResponseImpl>(kj::mv(paf.fulfiller));
-  auto promise = client->connect(address, *headers, *tunnel)
-      .attach(kj::str(address), kj::mv(headers));
-  tunnel->setPromise(kj::mv(promise));
-  kj::Promise<kj::HttpClient::ConnectResponse> connResponse = paf.promise.attach(kj::mv(tunnel));
-  return jsg::alloc<Socket>(kj::mv(connResponse));
+  auto httpClient = kj::newHttpClient(*client);
+  auto request = httpClient->connect(address, *headers);
+  auto revocable = kj::heap<RevocableIoStream>(*request.connection);
+
+  auto promise = request.status.then([&revocable](auto status) {
+    if (status.statusCode < 200 || status.statusCode >= 300) {
+      // The connection request failed!
+      // TODO(later): The status may include an errorBody that we currently do nothing with.
+      revocable->revoke(KJ_EXCEPTION(FAILED,
+          kj::str("jsg.Error: Unable to establish a TCP Socket (", status.statusCode,")")));
```

Comment:
The application doesn't actually know it's using HTTP, so telling it the HTTP status code seems at best weird and at worst could be leaking internal implementation details. We probably want to report a more meaningful error like "connection refused" or "DNS lookup failed" but that seems to require deep understanding of the errors returned by the proxy...

Reply:
Removed the statusCode from the error message and added a comment.